File: app/src/main/java/dev/olog/basil/presentation/main/VerticalPagerAdapter.java
package dev.olog.basil.presentation.main;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;
import dev.olog.basil.presentation.actions.SearchFragment;

public class VerticalPagerAdapter extends FragmentPagerAdapter {

    public VerticalPagerAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public Fragment getItem(int position) {
        if (position == 0){
            return new SearchFragment();
        }
        return new MainFragment();
    }

    @Override
    public int getCount() {
        return 2;
    }
}

File: presentation/src/main/java/dev/olog/basil/presentation/main/VerticalPagerAdapter.kt
package dev.olog.basil.presentation.main

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter
import dev.olog.basil.presentation.RecipeTypeChooserFragment
import dev.olog.basil.shared.throwNotHandled

class VerticalPagerAdapter(
        activity: FragmentActivity
) : FragmentStateAdapter(activity) {

    override fun getItemCount(): Int = 2

    override fun createFragment(position: Int): Fragment = when (position) {
        0 -> RecipeTypeChooserFragment()
        1 -> MainFragment()
        else -> throwNotHandled("view pager position=$position")
    }
}
